packages/react/src/jsx-namespace.ts

```diff
@@ -54,47 +54,53 @@ type ReactJSXIntrinsicElements = true extends IsPreReact19
   : /** @ts-ignore */
     React.JSX.IntrinsicElements
 
+// based on the code from @types/react@18.2.8
+// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3197efc097d522c4bf02b94e1a0766d007d6cdeb/types/react/index.d.ts#LL3204C13-L3204C13
+/** @ts-ignore */
 type ReactJSXElementType = true extends IsPreReact19
-  ? // based on the code from @types/react@18.2.8
-    // https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3197efc097d522c4bf02b94e1a0766d007d6cdeb/types/react/index.d.ts#LL3204C13-L3204C13
-    string | React.JSXElementConstructor<any>
+  ? string | React.JSXElementConstructor<any>
   : /** @ts-ignore */
     React.JSX.ElementType
 
 export namespace ReactJSX {
-  type ElementType = ReactJSXElementType
-  interface Element extends ReactJSXElement {}
-  interface ElementClass extends ReactJSXElementClass {}
-  interface ElementAttributesProperty
+  export type ElementType = ReactJSXElementType
```

Comment:
Am I correct in understanding that these two are equivalent? (since the former compiles into the latter)
 
```ts
export namespace Blah {
  export type Something = A
}
```

```ts
export declare namespace Blah {
  type Something = A
}
```

Reply:
loling at 

```ts
export namespace Blah {
  type A = {}
  export type Something = A
}
```
getting compiled into
```ts
export declare namespace Blah {
    type A = {};
    export type Something = A;
    export {};
}
```

(I know the context around `export {};`, just never thought about it for namespaces lol)

Reply:
Ye, those rules are super trippy tbh. I just slapped `export` onto those because they suddenly became unavailable when I moved this from an ambient content to `.ts` 🤷‍♂️